### PR TITLE
fix(Core/Item): Complimentary Brewfest Sampler

### PR DESCRIPTION
**Pull request by The-GhostRider**
## Changes Proposed:
This PR proposes changes to:
-  [x] Core (units, players, creatures, game systems).
-  [ ] Scripts (bosses, spell scripts, creature scripts).
-  [x] Database (SAI, creatures, etc).

- Inserted the aura in the "spell_custom_attr" Table
- Changed spell timer in "SpellInfoCorrections.cpp"

For further "Details" on how my fix operates, I recommend viewing my video: [[PR]: Conjured Complimentary Brewfest  (Fixed)](https://youtu.be/CsrdXFsePbE), which illustrates the functioning of the fix I implemented.

[Image N° 1]:

image

[Image N° 2]:

image

## Issues Addressed:

- Closes https://github.com/chromiecraft/chromiecraft/issues/4103
- Closes https://github.com/azerothcore/azerothcore-wotlk/issues/16986

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- - [[Report]: Conjured Complimentary Brewfest  (Bugged)](https://youtu.be/JKGfhKS3MZM) **[Before the Fix]**
- - [[PR]: Conjured Complimentary Brewfest  (Fixed)](https://youtu.be/CsrdXFsePbE) **[After the Fix]**
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: data/sql/updates/pending_db_world/rev_1691843684176706600.sql [hunk 1]
```diff
@@ -0,0 +1,4 @@
+--
+DELETE FROM `spell_dbc` WHERE (`ID` = 42533);
```
Comment (Kitzunu) [inline data/sql/updates/pending_db_world/rev_1691843684176706600.sql:2]: This is wrong, we dont fix spells in this table anymore. It is just for serverside spells not existing in the dbc files.

For any dbc corrections you do them in SpellInfoCorrections.cpp

File: src/server/game/Entities/Player/PlayerStorage.cpp [hunk 2]
```diff
@@ -7206,13 +7206,21 @@ void Player::_SaveAuras(CharacterDatabaseTransaction trans, bool logout)
     stmt->SetData(0, GetGUID().GetCounter());
     trans->Append(stmt);
 
+    // Add here the auras IDs to exclude
+    std::unordered_set<uint32> excludedAuraIds = {42533};
```
Comment (Kitzunu) [inline src/server/game/Entities/Player/PlayerStorage.cpp:7210]: And dont hardcode things

Comment (Nyeriah) [reply src/server/game/Entities/Player/PlayerStorage.cpp:7210]: There’s a spell attribute to skip saving auras afaik

**Comment by Nyeriah** [comment]
Unfortunatelly, we cannot accept fixes that rely on patching the client therefore this cannot be accepted as is.

**Comment by Nyeriah** [comment]
Reading through the issue, the spell attribute that prevents spells from being saved to the database should be able to address this without requiring any further patches

**Comment by The-GhostRider** [comment]
> Reading through the issue, the spell attribute that prevents spells from being saved to the database should be able to address this without requiring any further patches

image

**Comment by Nyeriah** [comment]
SPELL_ATTR0_CU_AURA_CANNOT_BE_SAVED

https://www.azerothcore.org/wiki/spell_custom_attr

**Comment by The-GhostRider** [comment]
> SPELL_ATTR0_CU_AURA_CANNOT_BE_SAVED
> 
> https://www.azerothcore.org/wiki/spell_custom_attr

I'm Testing it, I'll keep you up to date..

**Comment by The-GhostRider** [comment]
> SPELL_ATTR0_CU_AURA_CANNOT_BE_SAVED
> 
> https://www.azerothcore.org/wiki/spell_custom_attr

Ok, perfect, I wasn't aware of this new table, and I realized **(now looking at the entire DB)** that there are new tables I didn't know existed, I'll study them better.

Anyway, I put the ID with the right Flag and it works exactly the way it did with my C++ code, so that's fine too, if I had known this table existed I wouldn't have coded in C++.

**Review by Kitzunu — COMMENTED** [review]
bro you deleted the entire file lmao

File: data/sql/updates/pending_db_world/rev_1691843684176706600.sql (hunk 1 above)
Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1691843684176706600.sql:6]: restore PlayerStorage.cpp file and move the spell_dbc stuff into SpellInfoCorrections.cpp and this should be okay

**Review by Kitzunu — CHANGES_REQUESTED** [review]

**Comment by The-GhostRider** [comment]
> and move the spell_dbc stuff into SpellInfoCorrections.cpp and this should be okay

I'm sorry, but I disagree with this "Strategy" (if we want to call it that).

A METHODOLOGY/FUNCTION already exists for making these changes, and they are found within the DBC (DataBase Client).
In fact, the error is right there (at the root).
When you use C++ to make these changes.
By entering the data directly into the CODE, YOU make "HARDCODE".
Because the data is inside the code, and not inside the designed place (which is the DBC).

We always recommend using the enUS Client; this is because the "enUS Client" is a Client that is "less buggy" than the others, BUT! it's still buggy.

So, the error is in there, inside the DBCs.
If you use (A FOREIGN SOURCE) like C++ to (FORCE) the behavior of the DBs, you indeed "Force" a "pre-existing" management system.
And that's not wrong, but it's VERY WRONG!

**Comment by Kitzunu** [comment]
Like it or not, it's how we do it.

If you dont want to, we cant accept the PR.

**Comment by The-GhostRider** [comment]
> Like it or not, it's how we do it.
> 
> If you dont want to, we cant accept the PR.

ChatGPT also agrees with me.

I mean..
It's not like it's probative evidence.
but in short, its analysis adheres to mine.

image

**Comment by Kitzunu** [comment]
So create your own core.

This is how it is done in AC, so kindly, we dont care what chatgpt says.

Change it or close it. Trying to argue wont win you anything

**Comment by The-GhostRider** [comment]
> change it or close it.

I will change it by tomorrow; right now, I have things to do.

**Comment by JonathanAkaJD** [comment]
I'm flabbergasted at the self justification through chatbot. Gotta say, that's a first for me. You know the Adam Sandler movie, big daddy? Reminds me of the kid going "I wipe my own a**" Thanks for the laugh tho.

**Comment by The-GhostRider** [comment]
> I'm flabbergasted at the self justification through chatbot. Gotta say, that's a first for me. You know the Adam Sandler movie, big daddy? Reminds me of the kid going "I wipe my own a**" Thanks for the laugh tho.

You see...
Mine is not a justification (as you defined it).
I presented my (OPINION), based on valid analytical/logical arguments.

The fact that I consulted an A.I. to see what its "Analysis" of the matter was.
It was a simple "Pure" feedback that I shared with you, from an "A.I" which (unlike a human being), has an analytical (IMPARTIAL) opinion and is not subject to "Beliefs," "Likes and Dislikes," and the shallowness of the human psyche that you have displayed in this comment.

That was my entire intent, to have an impartial opinion.

**Comment by elthehablo** [comment]
> > Like it or not, it's how we do it.
> > If you dont want to, we cant accept the PR.
> 
> ChatGPT also agrees with me.
> 
> I mean.. It's not like it's probative evidence. but in short, its analysis adheres to mine.
> 
> image

image

**Review by Kitzunu — APPROVED** [review]

**Pull request merged**